include/librats/conf.h

```diff
@@ -35,6 +44,7 @@ typedef struct rats_conf {
 	char crypto_type[32];
 	rats_key_algo_t key_algo;
 	rats_hash_algo_t hash_algo;
+	user_defined_param user_param;
```

Comment:
Some fields in `user_defined_param` and `rats_conf structures` are repeated. Maybe we can directly use rats_conf to save the user defined param?